src/lib.rs

```diff
@@ -371,41 +346,31 @@ pub(crate) unsafe fn rav1d_open(c_out: &mut *mut Rav1dContext, s: &Rav1dSettings
         rav1d_refmvs_init(&mut (*f).rf);
         n = n.wrapping_add(1);
     }
-    let mut m: c_uint = 0 as c_int as c_uint;
-    while m < (*c).n_tc {
-        let t: *mut Rav1dTaskContext = &mut *((*c).tc).offset(m as isize) as *mut Rav1dTaskContext;
-        (*t).f = &mut *((*c).fc).offset(0) as *mut Rav1dFrameContext;
-        let thread_data = Arc::new(Rav1dTaskContext_task_thread::new(Arc::clone(
-            &(*c).task_thread,
-        )));
-        (&mut (*t).task_thread as *mut Arc<Rav1dTaskContext_task_thread>)
-            .write(Arc::clone(&thread_data));
-        *BitDepth16::select_mut(&mut (*t).cf) = Align64([0; 32 * 32]);
-        let handle = if (*c).n_tc > 1 as c_uint {
-            let thread_args = Box::new(Rav1dTaskContext_borrow {
-                c: &*c,
-                tc: &mut *t,
-            });
-            let mut thread = 0;
-            if pthread_create(
-                &mut thread,
-                &mut thread_attr,
-                Some(rav1d_worker_task),
-                Box::into_raw(thread_args).cast(),
-            ) != 0
-            {
-                return error(c, c_out, &mut thread_attr);
+    (*c).tc = (0..(*c).n_tc)
+        .map(|_| {
+            let thread_data = Arc::new(Rav1dTaskContext_task_thread::new(Arc::clone(
+                &(*c).task_thread,
+            )));
+            if (*c).n_tc > 1 {
+                let context_borrow = &*c;
```

Comment:
This is something that can be removed once `c` is no longer a raw pointer, right?  If so, could you leave a `TODO`?